pkg/component/status/status.go

```diff
@@ -0,0 +1,91 @@
+package status
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/k0sproject/k0s/internal/util"
+	"github.com/k0sproject/k0s/pkg/install"
+	"github.com/sirupsen/logrus"
+)
+
+type Status struct {
+	StatusInformation install.K0sStatus
+	Socket            string
+	L                 *logrus.Entry
+	address           string
+	httpserver        http.Server
+	listener          net.Listener
+}
+
+// Healthy dummy implementation
+func (s *Status) Healthy() error { return nil }
+
+// Init initializes component
+func (s *Status) Init() error {
+	s.L = logrus.WithFields(logrus.Fields{"component": "status"})
+
+	if len(s.address) == 0 {
+		s.address = path.Join(s.StatusInformation.K0sVars.RunDir, s.Socket)
+	}
+
+	if util.FileExists(s.address) {
+		return fmt.Errorf("%s already exists, use 'status-socket' to specify different unix socket", s.address)
+	}
+
+	var err error
+	s.httpserver = http.Server{
+		Handler: &statusHandler{Status: s},
+	}
+	err = util.InitDirectory(s.StatusInformation.K0sVars.RunDir, 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create %s: %w", s.address, err)
+	}
+
+	s.listener, err = net.Listen("unix", s.address)
+	if err != nil {
+		s.L.Errorf("failed to create listener %s", err)
+		return err
+	}
+	s.L.Infof("Listening address %s", s.address)
+
+	return nil
+}
+
+// Run runs the component
+func (s *Status) Run() error {
+	spew.Dump(s)
+	go func() {
+		if err := s.httpserver.Serve(s.listener); err != nil {
+			s.L.Errorf("failed to start status server at %s: %s", s.address, err)
+		}
+	}()
+	return nil
+}
+
+// Stop stops status component and removes the unix socket
+func (s *Status) Stop() error {
+	if err := s.httpserver.Shutdown(context.TODO()); err != nil {
+		return err
+	}
+	os.Remove(s.address)
```

Comment:
This should be `defer'd, otherwise a failure in shutdown will leave this lingering.